core/services/ocr2/plugins/ocr2keeper/log_provider.go

```diff
@@ -134,7 +319,148 @@ func (c *LogProvider) unmarshalLogs(logs []logpoller.Log) ([]performed, error) {
 	return results, nil
 }
 
+func (c *LogProvider) unmarshalReorgUpkeepLogs(logs []logpoller.Log) ([]reorged, error) {
+	results := []reorged{}
+
+	for _, log := range logs {
+		rawLog := log.ToGethLog()
+		abilog, err := c.registry.ParseLog(rawLog)
+		if err != nil {
+			return results, err
+		}
+
+		switch l := abilog.(type) {
+		case *registry.KeeperRegistryReorgedUpkeepReport:
+			if l == nil {
+				continue
+			}
+
+			r := reorged{
+				Log:                               log,
+				KeeperRegistryReorgedUpkeepReport: *l,
+			}
+
+			results = append(results, r)
+		}
+	}
+
+	return results, nil
+}
+
+func (c *LogProvider) unmarshalStaleUpkeepLogs(logs []logpoller.Log) ([]staleUpkeep, error) {
+	results := []staleUpkeep{}
+
+	for _, log := range logs {
+		rawLog := log.ToGethLog()
+		abilog, err := c.registry.ParseLog(rawLog)
+		if err != nil {
+			return results, err
+		}
+
+		switch l := abilog.(type) {
+		case *registry.KeeperRegistryStaleUpkeepReport:
+			if l == nil {
+				continue
+			}
+
+			r := staleUpkeep{
+				Log:                             log,
+				KeeperRegistryStaleUpkeepReport: *l,
+			}
+
+			results = append(results, r)
+		}
+	}
+
+	return results, nil
+}
+
+func (c *LogProvider) unmarshalInsufficientFundsUpkeepLogs(logs []logpoller.Log) ([]insufficientFunds, error) {
+	results := []insufficientFunds{}
+
+	for _, log := range logs {
+		rawLog := log.ToGethLog()
+		abilog, err := c.registry.ParseLog(rawLog)
+		if err != nil {
+			return results, err
+		}
+
+		switch l := abilog.(type) {
+		case *registry.KeeperRegistryInsufficientFundsUpkeepReport:
+			if l == nil {
+				continue
+			}
+
+			r := insufficientFunds{
+				Log: log,
+				KeeperRegistryInsufficientFundsUpkeepReport: *l,
+			}
+
+			results = append(results, r)
+		}
+	}
+
+	return results, nil
+}
+
+// Fetches the checkBlockNumber for a particular transaction and an upkeep ID. Requires a RPC call to get txData
+// so this function should not be used heavily
+func (c *LogProvider) getCheckBlockNumberFromTxHash(txHash common.Hash, id plugintypes.UpkeepIdentifier) (bk plugintypes.BlockKey, e error) {
```

Comment:
wanted to create unit test for this but it's heavyweight since there's no existing tests for logProvider so punting on it for now :(